future/src/ct/ct_misc_utils.h

```diff
@@ -439,4 +445,51 @@ std::string get_cherrytree_lang_filepath();
 
 std::string download_file(const std::string& filepath);
 
+/**
+ * @class path
+ * @brief An object representing a filepath
+ */
+class path {
+    using path_type = std::string;
+public:
+    path(path_type path) : _path(std::move(path)) {}
+    path(const char* path) : _path(path) {}
+    
+    ~path() = default;
+    
+    void append(const path& other) {
+        _path = Glib::build_filename(_path, other._path);
+    }
+
+    path& operator=(std::string other) {
+        _path.swap(other);
+        return *this;
+    }
+
+
+    friend path operator/(const path& lhs, const path& rhs) {
+        return path(Glib::build_filename(lhs._path, rhs._path));
+    }
+
+    friend path operator/(const path& lhs, const std::string& rhs) {
+        return path(Glib::build_filename(lhs._path, rhs));
+    }
+
+    friend path operator+(const path& lhs, const path& rhs) { return path(lhs._path + rhs._path); }
+    
+    const char* c_str() const { return string().c_str(); };
+    std::string string() const { return get_proper_platform_filepath(_path); }
```

Comment:
Is .string() used from std::filesystem? Otherwise I would rather prefer .str() being shorter

Reply:
Yes `string()` is from `std::filesystem`